✨ feat: Order 업데이트 API 구현

ssohn3 opened the pull request:
✨ feat: Order 업데이트 API 구현

- Order 주문 취소 api 구현( 배송 status, 주문 status) completed -> canclled 로 업데이트

Commit:
✨ feat: Order 업데이트 API 구현

Commit:
✨feature: Order UPDATE API 삭제 및 DELETE API 수정

ssohn3 commented:
@myoungjinGo-FE  삭제 API를 호출시 db에서 값을 order_info 를 삭제하지않고, cancellation 시간을 주문 취소 시간을 넣어주고, order status, shipping_status를 "canceled" 로 업데이트하게 만들었습니다.

Commit:
✨feature: Order UPDATE API 다시 생성 및 DELETE API 수정

Commit:
✨ feat: Create Order 수정

Order 주문 생성 할 경우 cancellation_date NULL로 저장 및 테스트 케이스 수정

myoungjinGo-FE commented on payments/api/views.py:
```diff
@@ -356,9 +357,26 @@ def order_detail(request, user_id, order_id):
             )
             return Response(response_data, status=status.HTTP_200_OK)
 
+        elif request.method == "PATCH":
+            # 결제 취소 업데이트 처리
+            order.payment.status = "canceled"
+            order.shipping_status = "canceled"
+            order.status = "canceled"
+            order.payment.save()
+            order.save()
+
+            logger.info(
+                f"Order 결제 취소 업데이트 성공 - order_id: {order_id}, user_id: {user_id}"
+            )
+            return Response(
+                {"message": "Order payment has been successfully updated."},
+                status=status.HTTP_200_OK,
+            )
```
이렇게 하는게 아니구
`PATCH users/{user_id}/orders/{order_id} data = {status:"canceled", shipping_status="canceled", status="canceled"} ` 와 같이 order의 속성값을 직접 입력받아야 합니다.
만약에 이렇게 받는게 비효율적이라고 생각이 된다면 `PATCH users/{user_id}/orders/{order_id}/cancel` 과 같이 어떠한 동작이 예상되도록 uri를 수정해주세요

ssohn3 replied:
url 수정했습니다.

myoungjinGo-FE commented on payments/tests.py:
```diff
@@ -383,92 +376,108 @@ def test_결제_실패_서비스_확인(test_유저):
     )
     response = client.delete(url)
 
+    # 주문이 취소 처리가 되었는지 확인
+    cancled_order = Order.objects.get(id=order_id)
+
+    # cancellation_date가 기록되었는지 확인
+    assert cancled_order.cancellation_date is not None
+
+    # 응답이 성공적인지 확인
     assert response.status_code == 200
     assert response.json() == {"message": "Order has been successfully deleted."}
```
조회 API를 호출해서 `cancellation_date`가 None이 아닌 order를 가져오지 않는지 테스트해야 할거 같습니다.

ssohn3 replied:
테스트 수정했습니다.

Commit:
✨ feat: Order PATCH, DELETE API 수정

- Order 주문 취소 api PATCH users/{user_id}/orders/{order_id}/cancel 로 변경했습니다.

- test_주문_취소 테스트 코드 수정: 조회 API 호출해서 order를 가져오지 않는지 테스트 하도록 변경했습니다.

myoungjinGo-FE commented on payments/tests.py:
```diff
@@ -447,25 +447,29 @@ def test_주문_취소(client, test_유저, test_주문서_생성):
     assert response.json() == {"message": "Order has been successfully deleted."}
 
+    # 삭제 후 주문 조회하여 cancellation_date 확인
+    response_get = client.get(url)
+    assert response_get.status_code == 200
+
+    # 조회된 주문 데이터에서 cancellation_date 확인
+    cancled_order_data = response_get.json()
+    assert cancled_order_data["order_info"]["cancellation_date"] is not None
```
삭제된 데이터는 불러올 수 없어야 합니다.

ssohn3 replied:
안 불러오게 만들었습니다.

myoungjinGo-FE commented on payments/api/serializers.py:
```diff
@@ -40,4 +40,5 @@ class Meta:
             "shipping_status",
             "products",
             "status",
+            "cancellation_date",
```
다시 빼도 될거같습니다.

ssohn3 replied:
제거 했습니다.

Commit:
✨ feat: Order PATCH, DELETE API 수정

- Order 주문 내역 취소 api를 요청하면 주문 내역에서 해당 주문을 불러 올 수 없도록 수정했습니다

- test_주문_취소 테스트 코드 수정

myoungjinGo-FE commented on payments/api/views.py:
```diff
@@ -332,6 +332,14 @@ def order_detail(request, user_id, order_id):
+                        "error": f"Order with id '{order_id}' for user '{user_id}' not found."
+                    },
+                    status=status.HTTP_404_NOT_FOUND,
+                )
```
이렇게 해도 돌아는 가는데 그냥 objects.get 여기서 filtering을 해도 되지 않을까.................싶지만 네

ssohn3 replied:
수정했습니다

myoungjinGo-FE commented:
OK 좋습니다.

Commit:
✨ feat: Object.get에서 예외처리로 인해 주문 내역이 취소된거면 호출하지 않게 수정했습니다.

Commit:
✨ feat: 수정
